File: 백준/Gold/2230. 수 고르기/수 고르기.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
})

const input = [];
const numberList = [];

rl.on('line', (line) => {
    input.push(line);
}).on('close', () => {
    const [n, m] = input[0].split(' ').map(Number);

    for (let i = 1; i <= n; i++) {
        numberList.push(Number(input[i]));
    }

    let minD = 2000000000;
    let left = 0;
    let right = 1;

    numberList.sort((a, b) => a - b); // 정렬

    while (right < n) {
        const diff = numberList[right] - numberList[left];

        if (diff >= m) {
            minD = Math.min(minD, diff);
            left++;
        } else {
            right++;
        }
    }

    console.log(minD);
})

File: 백준/Gold/2230. 수 고르기/수 고르기.py
n, m = map(int, input().split())

arr = [int(input()) for _ in range(n)]
arr.sort()

left = 0
right = 0
result = 2_000_000_000

while right < n:
    if arr[right] - arr[left] >= m:
        result = min(result, arr[right] - arr[left])
        left+= 1
    else:
        right += 1
    
    if left > right:
        right = left

print(result)